File: services/payment-service/src/main/java/com/banking/platform/paymentservice/domain/model/Payment.java
package com.banking.platform.paymentservice.domain.model;
import lombok.*;
import java.math.BigDecimal;
import java.time.Instant;
import java.util.UUID;

@Getter @Builder @AllArgsConstructor @NoArgsConstructor
public class Payment {
    private UUID id;
    private UUID customerId;
    private UUID accountId;
    private String payee;
    private String payeeAccount;
    private BigDecimal amount;
    private String currency;
    private PaymentType type;
    private PaymentStatus status;
    private String reference;
    private String idempotencyKey;
    private Instant createdAt;
    private Instant confirmedAt;

    public enum PaymentType { BILL_PAYMENT, P2P_TRANSFER, MERCHANT_PAYMENT, INTERNAL_TRANSFER }
    public enum PaymentStatus { INITIATED, PROCESSING, CONFIRMED, FAILED, REFUNDED }
}

File: frontend/src/types/payment.ts
import { PaymentType, PaymentStatus } from './enums';

export interface Payment {
  id: string;
  customerId: string;
  accountId: string;
  payee: string;
  payeeAccount: string;
  amount: number;
  currency: string;
  type: PaymentType;
  status: PaymentStatus;
  reference: string;
  createdAt: string;
  confirmedAt?: string;
}

export interface CreatePaymentRequest {
  customerId: string;
  accountId: string;
  payee: string;
  payeeAccount: string;
  amount: number;
  currency: string;
  type: PaymentType;
  description?: string;
  idempotencyKey: string;
}
